src/trade.js

```diff
@@ -12,6 +12,15 @@ class Trade {
     this._expiresAt = null;
     this._delegate = delegate;
     this._api = api;
+    this._fromApi = false;
+
+    if (obj !== null) {
+      this._id = obj.id;
+    }
+  }
+
+  setFromAPI () {
+    this._fromApi = true;
```

Comment:
There's two sources of data for the Trade class. One is the KV store `trades` array, the other is the exchange specific API. The API contains much more - and up to date - information than the KV store. 

The `Trade` objects are created at the same time as the `Exchange` object. The API info is only loaded when you call `.getTrades()`. Meanwhile functions like `watchAddress()` will work fine even before you call `getTrades()`, which is good for privacy: it means we can monitor for buy order completion each time the user logs into their wallet, without pinging the partner API.

At the moment `_fromApi` is not exposed publicly and not actually used for anything. It should probably only be used to throw errors if you try to call a method too early, or perhaps to automatically call `getTrades()` or `refresh()` first, for methods that depend on information from the API.